src/utils/schema/setAddress.ts

```diff
@@ -1,12 +1,18 @@
 import { Address } from 'src/types';
 
-export function setAddress(address?: Address) {
-  if (address) {
-    return {
-      '@type': 'PostalAddress',
-      ...address,
-    };
-  }
-
-  return undefined;
+export function setAddress(address?: Address | Address[]) {
+  if (!address) return undefined;
+
+  // If array of one address, replace with single address
+  if (Array.isArray(address) && address.length === 1) address = address[0];
+
+  // If array, return mapped array of PostalAddresses
+  if (Array.isArray(address)) return address.map(toPostalAddress);
+
+  // Return single PostalAddress
+  return toPostalAddress(address);
+}
```

Comment:
Only comment would be I generally try to avoid re-assigning. If we were to re-write something like this we could cut out a check or two and short circuit by returning early?

```suggestion
export function setAddress(address?: Address | Address[]) {
  if (!address) return undefined;
  
  if (!Array.isArray(address)) return toPostalAddress(address);

  // If array of one address, replace with single address
  if (address.length === 1) return toPostalAddress(address[0]);

  // If array, return mapped array of PostalAddresses
  return address.map(toPostalAddress);
}
```

Reply:
LGTM, ran it locally and all tests succeed 👍